src/main/java/macrobase/analysis/summary/Summarizer.java

```diff
@@ -1,30 +1,21 @@
 package macrobase.analysis.summary;
 
+import macrobase.analysis.pipeline.operator.MBConsumer;
+import macrobase.analysis.pipeline.operator.MBProducer;
 import macrobase.analysis.result.OutlierClassificationResult;
 import macrobase.analysis.classify.OutlierClassifier;
 import macrobase.conf.MacroBaseConf;
 import macrobase.ingest.DatumEncoder;
 
-import java.util.Iterator;
-
 /**
  * Consumes OutlierClassification Result tuple-at-a-time, but returns summaries
  # when there has been an update.
  */
-public abstract class Summarizer implements Iterator<Summary> {
 
-    protected final OutlierClassifier input;
+public abstract class Summarizer implements MBConsumer<OutlierClassificationResult>, MBProducer<Summary> {
```

Comment:
What do you think about making OutlierClassifier, FeatureTransform, and Summarizer all interfaces? It decouples the types from any shared implementation code and makes it harder for extra dependencies to sneak into the definition of a Summarizer so we'll always be able to write clean testing stubs.